test/preconditioner/batch_jacobi_kernels.cpp

```diff
@@ -0,0 +1,231 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/preconditioner/batch_jacobi_kernels.hpp"
+
+
+#include <limits>
+#include <random>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/batch_multi_vector.hpp>
+#include <ginkgo/core/base/exception.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/math.hpp>
+#include <ginkgo/core/matrix/batch_csr.hpp>
+#include <ginkgo/core/preconditioner/batch_jacobi.hpp>
+
+
+#include "core/test/utils.hpp"
+#include "core/test/utils/batch_helpers.hpp"
+#include "test/utils/executor.hpp"
+
+
+namespace detail {
+
+
+template <typename ValueType>
+void is_equivalent_to_ref(
+    std::unique_ptr<gko::batch::preconditioner::Jacobi<ValueType>> ref_prec,
+    std::unique_ptr<gko::batch::preconditioner::Jacobi<ValueType>> d_prec)
+{
+    auto ref = ref_prec->get_executor();
+    auto exec = d_prec->get_executor();
+    const auto nbatch = ref_prec->get_num_batch_items();
+    const auto num_blocks = ref_prec->get_num_blocks();
+    const auto block_pointers_ref = ref_prec->get_const_block_pointers();
+
+    const auto& ref_storage_scheme = ref_prec->get_blocks_storage_scheme();
+    const auto& d_storage_scheme = d_prec->get_blocks_storage_scheme();
+
+    const auto tol = 100000 * r<ValueType>::value;
```

Comment:
wow, that is quite tolerant. I'm not sure if this will have usefulness for float.